python/cudf_polars/cudf_polars/testing/asserts.py

```diff
@@ -117,12 +121,23 @@ def assert_ir_translation_raises(q: pl.LazyFrame, *exceptions: type[Exception])
     AssertionError
        If the specified exceptions were not raised.
     """
+    translator = Translator(q._ldf.visit())
     try:
-        _ = translate_ir(q._ldf.visit())
-    except exceptions:
-        return
+        _ = translator.translate_ir()
     except Exception as e:
-        raise AssertionError(f"Translation DID NOT RAISE {exceptions}") from e
+        # The only ways an exception is raised in translation is if
+        # ...
+        if (
+            "No support for polars IR" not in str(e)
+            and "Could not compute schema" not in str(e)
+            and "Could not retrieve the current" not in str(e)
+        ):
+            raise IRTranslationFailed("assert_ir_translation_raises failed") from e
```

Comment:
Does `translate_ir` now not catch and re-raise these exceptions? Perhaps it should?